cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -237,13 +239,21 @@ enum class FailureReason {
     HIT_EARLY_TERMINATION,
 }
 
+/**
+ * This data class represents a path of nodes as some paths are not just lists of edges, but instead
+ * can be traversed by a predicate.
+ */
+data class NodePath(val nodes: List<Node>) : HasAssumptions {
+    override val assumptions: MutableList<Assumption> = mutableListOf()
```

Comment:
Can you move this directly to the constructor?

Reply:
Done

Reply:
Wait. As you didn't want to use this in the other PR, it doesn't make sense to introduce it here either